Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/profile_page.tsx

```diff
@@ -44,14 +58,737 @@ export class ProfilePage extends React.Component<Properties> {
+                    value={this.props.identity.province}
+                    displaySize={this.props.displaySize}
+                    disabled/>
+                </FormEntry>
+                <Dali.Padding size={ProfilePage.LINE_PADDING}/>
+                <HLine color={ProfilePage.LINE_COLOR}/>
+                <Dali.Padding size={ProfilePage.LINE_PADDING}/>
+                <FormEntry name='Country'
+                  orientation={orientation}>
+                  <CountrySelectionBox
+                    readonly
+                    displaySize={this.props.displaySize}
+                    value={this.props.identity.country}
+                    countryDatabase={this.props.countryDatabase}/>
+                </FormEntry>
+                <Dali.Padding size={ProfilePage.STANDARD_PADDING}/>
+                {formFooter}
+                <Dali.Padding size={formFooterPaddingSize}/>
+              </Dali.VBoxLayout>
+            </Dali.HBoxLayout>
+            <SubmitCommentBox displaySize={this.props.displaySize}
+              hasError={this.props.hasError}
+              submitStatus={this.props.submitStatus}/>
+            <Dali.Padding size={ProfilePage.STANDARD_PADDING}/>
+            <HLine color={ProfilePage.LINE_COLOR}/>
+            <Dali.Padding size={ProfilePage.STANDARD_PADDING}/>
+            <ChangePasswordBox displaySize={this.props.displaySize}
+              hasPasswordError={this.props.hasPasswordError}
+              submitPasswordStatus={this.props.submitPasswordStatus}/>
+            <Dali.Padding size={ProfilePage.BOTTOM_PADDING}/>
+          </Dali.VBoxLayout>
+        </div>
+        <div style={ProfilePage.STYLE.pagePadding}/>
+      </div>);
+  }
+
+  private static readonly STYLE = {
+    page: {
+      height: '100%',
+      width: '100%',
+      display: 'flex' as 'flex',
+      flexDirection: 'row' as 'row',
+      justifyContent: 'space-between' as 'space-between',
+      overflowY: 'auto' as 'auto'
+    },
+    hidden: {
+      opacity: 0,
+      visibility: 'hidden' as 'hidden',
+      display: 'none' as 'none'
+    },
+    headerStyler: {
+      color: '#4B23A0',
+      font: '500 14px Roboto',
+      width: '100%'
+    },
+    lastLoginBox: {
+      color: '#000000',
+      font: '400 14px Roboto'
+    },
+    contentSmall: {
+      flexGrow: 1,
+      maxWidth: '424px'
+    },
+    contentMedium: {
+      width: '732px'
+    },
+    contentLarge: {
+      width: '1000px'
+    },
+    pagePadding: {
+      width: '30px'
+    },
+    rolesWrapper: {
+      marginLeft: '11px',
+      display: 'flex' as 'flex',
+      flexDirection: 'row' as 'row',
+      height: '34px',
+      justifyContent: 'flex-start',
+      alignItems: 'center'
+    }
+  };
+  private static readonly LINE_PADDING = '14px';
+  private static readonly STANDARD_PADDING = '30px';
+  private static readonly BOTTOM_PADDING = '60px';
+  private static readonly LINE_COLOR = '#E6E6E6';
+}
+
+interface SubmitCommentBoxProperties  {
```

Comment:
Put these components into their own file and import them. It's too big/long to have them all in a single file.